app/js/enhancedDate.js

```diff
@@ -1,5 +1,107 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+var enhancedDate = (function iffe() {
+  'use strict';
 
+  //variables for all of the functions below declared here
+
+
+  var myDate;
+  var myDayName;
+  var dayNames = ["Sunday", "Monday","Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
+  var myMonth;
+  var monthNames = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+  var todaysDate;
+
+
+  return {
+
+
+    setDate: function setDate(enterDate) {
+        if (enterDate instanceof Date){
+          myDate = enterDate;
+        } else if (isNaN(enterDate) === false){
+          myDate = enterDate;
+        } else{
+          myDate = new Date();
+        }
+    },
+
+
+    getDate: function getDate(enterDate) {
+         if (enterDate instanceof Date){
+            myDate = enterDate;
+         }else if (typeof enterDate === "number"){
+            myDate = enterDate;
+         }else{
+           myDate = new Date();
+         }
+         return myDate;
+
+    },
+
+    getDayName: function getDayName(enterDate) {
+
+      if (enterDate instanceof Date){
+         myDate = enterDate;
+      }else if (typeof enterDate === "number"){
+         myDate = enterDate;
+      }else{
+        myDate = new Date();
+      }
+      myDayName = dayNames[myDate.getDay()];
+      return myDayName;
+    },
+
+    getMonthName: function getMonthName(enterDate) {
+      if (enterDate instanceof Date){
+         myDate = enterDate;
+      }else if (typeof enterDate === "number"){
+         myDate = enterDate;
+      }else{
+        myDate = new Date();
+      }
+      myMonth = monthNames[myDate.getMonth()];
+      return myMonth;
+    },
+
+    isFuture: function isFuture(enterDate) {
```

Comment:
This function doesn't take any arguments, it just uses the date you should already have stored. Now that I think about it, I think the confusion here has to do with the logic of checking whether you already have a Date object set or not. What you want to do for that is to check if the existing myDate variable is set or not (you can check if it is already an instanceof Date, and if it isn't, set it to a new instance of Date. Put that check and instantiation of a new Date into its own function, and each of these functions that you're passing enterDate into would just call that method rather than using a parameter.